faker/providers/misc/__init__.py

```diff
@@ -130,12 +130,17 @@ def locale(self):
     def language_code(self):
         return self.random_element(Provider.language_locale_codes.keys())
 
-    def uuid4(self):
+    def uuid4(self, cast_to=str):
         """
         Generates a random UUID4 string.
+        :param cast_to: Specify what type the UUID should be cast to. Default is `str`
+        :type cast_to: callable, None
         """
         # Based on http://stackoverflow.com/q/41186818
-        return str(uuid.UUID(int=self.generator.random.getrandbits(128)))
+        value = uuid.uuid4(int=self.generator.random.getrandbits(128))
```

Comment:
sorry, my bad. `uuid4` doesn't support the `int` argument. Can you change this line to `value = uuid.UUID(int=self.generator.random.getrandbits(128), version=4)`?